tower-http/src/services/fs/serve_dir.rs

```diff
@@ -120,6 +124,16 @@ impl ServeDir {
         self.buf_chunk_size = chunk_size;
         self
     }
+
+    /// Cache-Control:max-age=300.
+    ///
+    /// This speeds up user's feeling for static sites, especially large files.
+    ///
+    /// Defaults to 300 seconds.
+    pub fn max_age(mut self, dur: Option<Duration>) -> Self {
```

Comment:
Actually I think this will be more ergonomic.

```suggestion
    pub fn max_age(mut self, max_age: impl Into<Option<Duration>>) -> Self {
```

Hyper does the same for configs.